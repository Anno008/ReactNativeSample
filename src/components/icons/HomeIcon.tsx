import React from "react";
import { View } from "react-native";
import { Path, Svg } from "react-native-svg";
import { IIconProps } from "./IIconProps";

export const HomeIcon: React.SFC<IIconProps> = (props: IIconProps): JSX.Element => (
    <View>
        <Svg width={props.size} height={props.size} viewBox="0 0 24 24">
            <Path fill="none" d="M0 0h24v24H0V0z" />
            <Path fill={props.tintColor} d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z" />
        </Svg>
    </View>
);
