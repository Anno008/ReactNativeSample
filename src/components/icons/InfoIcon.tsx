import React from "react";
import { View } from "react-native";
import { Path, Svg } from "react-native-svg";
import { IIconProps } from "./IIconProps";

export const InfoIcon: React.SFC<IIconProps> = (props: IIconProps) => (
  <View>
    <Svg width={props.size} height={props.size} viewBox="0 0 24 24">
      <Path fill="none" d="M0 0h24v24H0V0z" />
      <Path
        fill={props.tintColor}
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </Svg>
  </View>
);
