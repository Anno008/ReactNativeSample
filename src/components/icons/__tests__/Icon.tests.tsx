import { shallow } from "enzyme";
import React from "react";
import { HomeIcon } from "../HomeIcon";
import { InfoIcon } from "../InfoIcon";

describe("Icon tests", (): void => {
    it("Should match HomeIcon snapshot", (): void => {
        const wrapper = shallow(<HomeIcon size={20} tintColor="blue" />);

        expect(wrapper).toMatchSnapshot();
    });

    it("Should match InfoIcon snapshot", (): void => {
        const wrapper = shallow(<InfoIcon size={20} tintColor="blue" />);

        expect(wrapper).toMatchSnapshot();
    });
});
