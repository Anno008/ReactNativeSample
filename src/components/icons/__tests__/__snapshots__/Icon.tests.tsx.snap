// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon tests Should render HomeIcon 1`] = `ShallowWrapper {}`;

exports[`Icon tests Should render InfoIcon 1`] = `ShallowWrapper {}`;
